inspirehep/modules/multieditor/actions.py

```diff
@@ -0,0 +1,299 @@
+            else:
+                self.apply_action(record[key], new_schema, position + 1, checked)
+
+
+def create_schema_record(schema, path, value):
+    """Object creation in par with the schema."""
+    record = {}
+    temp_record = record
+    new_schema = schema
+    if new_schema['type'] == 'array':
+        new_schema = new_schema['items']['properties']
+    elif new_schema['type'] == 'object':
+        new_schema = new_schema['properties']
+    for key in path:
+        new_schema = new_schema[key]
+        if new_schema['type'] == 'object':
+            new_schema = new_schema['properties']
+            temp_record[key] = {}
+            temp_record = temp_record[key]
+
+        elif new_schema['type'] == 'array':
+            if new_schema['items']['type'] == 'object':
+                new_schema = new_schema['items']['properties']
+            if len(path) == 1:
+                temp_record[key] = [value]
+                return record
+            else:
+                temp_record[key] = [{}]
+                temp_record = temp_record[key][0]
+    temp_record[path[-1]] = value
+    return record
+
+
+def check_value(record, match_type, keys, value_to_check, position):
+    """Where continues to find the value."""
+    key = keys[position]
+    if not record.get(key):
+        if match_type == 'does not exist':
+            return True
+        else:
+            return False
+    temp_record = record[key]
+    if isinstance(temp_record, list):
+        for index, array_record in enumerate(temp_record):
+            if position + 1 == len(keys):
+                if match_type == 'is equal to' and\
+                        array_record == value_to_check:
+                    return True
+                elif match_type == 'contains' and\
+                        value_to_check in array_record:
+                    return True
+                elif match_type == 'matches regular expression'and\
+                        re.search(
+                            re.escape(value_to_check),
+                            array_record):
+                            return True
+                elif match_type == 'does not exist' and\
+                        value_to_check == array_record:
+                    return True
+            else:
+                if check_value(array_record, match_type,
+                               keys, value_to_check, position + 1):
+                    return True
+    else:
+        if position + 1 == len(keys):
+            if match_type == 'is equal to' and \
+                    temp_record == value_to_check:
+                return True
+            elif match_type == 'contains' and \
+                    value_to_check in temp_record:
+                return True
+            elif match_type == 'matches regular expression' and \
+                    re.search(
+                        re.escape(value_to_check),
+                        temp_record):
+                return True
```

Comment:
Just checking, `does not exist` does not make sense in the context of checking an object?

Reply:
Actually the check is being made at the start of the code, the code for "does not exist" was wrong in the array.